src/libraries/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -213,22 +214,161 @@ public static void ReadObjectFail(string json, bool failsOnObject)
         }
 
         [Fact]
-        public static void ReadObjectFail_ReferenceTypeMissingParameterlessConstructor()
+        public static void ReadObjectFail_ReferenceTypeMissingPublicParameterlessConstructor()
         {
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<PublicParameterizedConstructorTestClass>(@"{""Name"":""Name!""}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithInternalParameterlessCtor>(@"{""Name"":""Name!""}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithPrivateParameterlessCtor>(@"{""Name"":""Name!""}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<CollectionWithoutPublicParameterlessCtor>(@"[""foo"", 1, false]"));
+
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<GenericClassWithProtectedInternalCtor<string>>("{\"Result\":null}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ConcreteDerivedClassWithNoPublicDefaultCtor>("{\"ErrorString\":\"oops\"}"));
         }
 
-        class PublicParameterizedConstructorTestClass
+        public class PublicParameterizedConstructorTestClass
         {
-            private readonly string _name;
             public PublicParameterizedConstructorTestClass(string name)
             {
-                _name = name;
+                throw new InvalidOperationException();
```

Comment:
Does this actually get called? If not, use a `Debug.Assert(false)` (or fail the test)

Reply:
Alright, will change it to `Debug.Fail`.